packages/services/api/src/modules/audit-logs/providers/audit-log-recorder.ts

```diff
@@ -0,0 +1,80 @@
+import { Injectable, Scope } from 'graphql-modules';
+import { captureException } from '@sentry/node';
+import { type User } from '../../../shared/entities';
+import { sql as c_sql, ClickHouse } from '../../operations/providers/clickhouse-client';
+import { Logger } from '../../shared/providers/logger';
+import { auditLogSchema, type AuditLogEvent } from './audit-logs-types';
+
+type AuditLogRecordEvent = {
+  userId: string;
+  userEmail: string;
+  organizationId: string;
+  user: (User & { isAdmin: boolean }) | null;
+};
+
+@Injectable({
+  scope: Scope.Operation,
+  global: true,
+})
+/**
+ * Responsible for recording audit log events and storing them in ClickHouse
+ */
+export class AuditLogRecorder {
+  private logger: Logger;
+
+  constructor(
+    logger: Logger,
+    private clickHouse: ClickHouse,
+  ) {
+    this.logger = logger.child({ source: 'AuditLogRecorder' });
+  }
+
+  record(event: AuditLogEvent, ctx: AuditLogRecordEvent): void {
+    // Ensures audit log is fail-safe and non-blocking
+    void this.internalCreateLogAuditEvent(event, ctx);
```

Comment:
Why does it need to be unblocking?

Reply:
What if the audit log insert fails, do we need to gurantee that the audit log insertion succeds? Should we delay writing to a queue with retries?

Reply:
TODO @dotansimha please confirm these questions

Reply:
TODO @TuvalSimha: make method `record` async, await the insert and all `AuditLogRecorder.record` calls